presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -439,6 +429,23 @@ public synchronized void updatePartitionStatistics(HiveIdentity identity, Table
                 .collect(toImmutableList());
     }
 
+    @GuardedBy("this")
+    private List<String> listAllTables(String databaseName)
+    {
+        requireNonNull(databaseName, "databaseName is null");
+
+        Optional<Database> database = getDatabase(databaseName);
+        if (database.isEmpty()) {
+            return ImmutableList.of();
+        }
+
+        Path databaseMetadataDirectory = getDatabaseMetadataDirectory(databaseName);
```

Comment:
This can be inline
```java
return getChildSchemaDirectories(getDatabaseMetadataDirectory(databaseName)).stream()
        .map(Path::getName)
        .collect(toImmutableList());

Reply:
This is existing code and i prefer not to improve it this time.